pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1118,13 +1118,24 @@ func maybeRetryBoot(ctx *domainContext, status *types.DomainStatus) {
 	}
 	defer file.Close()
 
+	if err := hyper.Task(status).VirtualTPMSetup(status.DomainName, agentName, ctx.ps, warningTime, errorTime); err != nil {
+		log.Errorf("Failed to setup virtual TPM for %s: %s", status.DomainName, err)
+		status.VirtualTPM = false
```

Comment:
In theory, we can check it once and save it to status by calling `PublichDomainStatus()`, but I'm always confused about how it works. So, this approach is more robust, I would say =)

Reply:
yeah, I don't think it is necessary to publish it, the Terminate and Teardowns are no-op if the there is no vTPM instance with the given ID, so better safe than sorry.